domain/keyupdater/service/service.go

```diff
@@ -118,10 +128,30 @@ func (s *Service) GetAuthorisedKeysForMachine(
 		)
 	}
 
-	machineKeys, err := s.st.AuthorisedKeysForMachine(ctx, machineName)
+	err := s.st.CheckMachineExists(ctx, machineName)
```

Comment:
It doesn't need to be. This is just validation. We don't have any special logic on a per machine basis so this makes no sense at the moment.

Reply:
What happens if the machine exists right now, but in the moment of time taken to start the txn for run atomic, the machine is deleted. The assumption that the machine exists is now violated. I guess your point is that the machine is never used in the subsequent logic. If that's the case, why even pass in the machine name to start with? I guess as you say below that's the question the API is asking. But if machine is not even used, the API is asking the wrong question. If the policy is all keys for a user get put on all machines, and we don't have the modelling to restrict some machines not to get all keys, then this behaviour makes no sense right now. But if it's to preserve existing behaviour, then can we add a big comment to explain this since if we were writing this today, the API would make more sense as `GetAuthorisedKeysForUser` not machine.

Reply:
You can't do anything about it. We are split across two databases. You will never solve this problem. If the machine exists or doesn't exist at the time we get the keys is irrelevant as we will return the exact same answer. When and if we ever get that complex in the model then we can re-address this. But you can't have a tx over multiple DB's period!

Reply:
I forgot we were hitting the controller db for some reason.